kpm/lib/kpm/killbill_server_artifact.rb

```diff
@@ -61,14 +61,16 @@ def info(version = 'LATEST', sha1_file = nil, force_download = false, verify_sha
                               verify_sha1,
                               overrides,
                               ssl_verify)
-
-          pom = REXML::Document.new(File.new(oss_pom_info[:file_path]))
+          file = File.new(oss_pom_info[:file_path])
+          pom = REXML::Document.new(file)
           properties_element = pom.root.elements['properties']
           %w[killbill-api killbill-plugin-api killbill-commons killbill-platform].each do |property|
             versions[property] = properties_element.elements["#{property}.version"].text
+            file.close
```

Comment:
I believe you need the `close` outside of the `for` loop.

I would also suggest to do this in a `ensure` (equivalent to `finally`) block:

```
file = File.new(oss_pom_info[:file_path])
begin
  pom = REXML::Document.new(file)
  properties_element = pom.root.elements['properties']
  %w[killbill-api killbill-plugin-api killbill-commons killbill-platform].each do |property|
    versions[property] = properties_element.elements["#{property}.version"].text
  end
ensure
  file.close
  FileUtils.rm_f(oss_pom_info[:file_path])
end
```